server/src/session.rs

```diff
@@ -1,15 +1,51 @@
 // org.freedesktop.Secret.Session
 
-use zbus::interface;
+use std::sync::{Arc, Mutex};
+
+use oo7::Key;
+use zbus::{interface, zvariant::OwnedObjectPath};
 
 use super::Result;
+use crate::{consts::SECRET_SESSION_PREFIX, service_manager::ServiceManager};
 
-#[derive(Debug)]
-pub struct Session {}
+#[derive(Debug, Clone)]
+pub struct Session {
+    _aes_key: Option<Arc<Key>>,
+    manager: Arc<Mutex<ServiceManager>>,
+    path: OwnedObjectPath,
+}
 
 #[interface(name = "org.freedesktop.Secret.Session")]
 impl Session {
-    pub async fn close(&self) -> Result<()> {
-        todo!()
+    pub async fn close(
+        &self,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
+    ) -> Result<()> {
+        self.manager.lock().unwrap().remove_session(self.path());
+        object_server.remove::<Self, _>(&self.path).await?;
+
+        Ok(())
+    }
+}
+
+impl Session {
+    pub fn new(
+        aes_key: Option<Arc<Key>>,
+        manager: Arc<Mutex<ServiceManager>>,
+        session_counter: i32,
```

Comment:
You already take the manager, so why require also the counter? increment it here instead.

Reply:
I don't get it. The `session_counter` attribute is in the `Service` struct and not in the `ServiceManager`. Do you mean I should get the session count from `sessions` HashMap?

Reply:
My bad, I read too fast. Then rename session_counter to session_index.

Reply:
no worries, done :)